content/Watsonx/CodeAssistant/Z/112.md

```diff
@@ -0,0 +1,103 @@
+---
+title: '112: Application Discovery'
+timeToComplete: 15
+updated: 2024-03-14
+---
+
+<QuizAlert text='Heads Up! Quiz material will be flagged like this!' />
+
+# 112: IBM watsonx Code Assistant (WCA) for Z Application Discovery
+
+### Goal
+
+The goal of this lab is to focus on the **Understand** phase of the mainframe application modernization lifecycle for z/OS applications. Going through an application discovery process to understand all the dependencies and relationships in an application is an important first step in any application modernization journey. Architects and developers can spend weeks or months lost trying to understand all the changes needed by reviewing millions of lines of code, hundreds of dependencies, and dated documentation. Using an application discovery tool helps to discover the relationships between application components, and enables developers to visually analyze those relationships, so they can more quickly and safely make changes for a broad set of languages and environments.
+
+## Application Discovery
+
+1. Click on the **IBM Developer for z/OS** icon in the taskbar to open it.
+
+  ![](./images/112/open-app.png)
+
+2. A popup will appear asking you to select a directory as workspace. Keep the default directory and press the **Launch** button. **Note:** It might take a couple minutes to load.
+
+  ![](./images/112/launch-workspace.png)
+
+3. The **IBM Developer for z/OS** opens in the **Explore projects** tab. Right-click anywhere below the **quick filter** field and select **Get project list**.
+
+  ![](./images/112/get-projects-list.png)
+
+4. Select **GenApp** from the new populated **Project** column.
+
+  ![](./images/112/select-genapp.png)
+
+5. A new field with a list of options will appear on the right side of the screen with its own quick filter at the top. In this new field, expand the **Mainframe Graphs** twistie, if not expanded already, and double-click on the **Transaction Callgraph** option. **Note:** It may take a moment for the pop-up needed to do the next step to appear. You will see **executing Transaction Callgraph** at the bottom with a green bar.
```

Comment:
```suggestion
5. A new field with a list of options will appear on the right side of the screen with its own quick filter at the top. In this new field, expand the **Mainframe Graphs** twistie, if not expanded already, and double-click on the **Transaction Callgraph** option. 
> **Note:** It may take a moment for the pop-up needed to do the next step to appear. You will see **executing Transaction Callgraph** at the bottom with a green bar.
```